### DISCUSSION TITLE
Does the data collection opt-out also opt-out of "User Engagement Data"?

### DISCUSSION THREAD

**Discussion by Cattes** [community/community#20138] [Copilot Conversations] [answered]
If the "User Engagement Data" is collected despite the data collection opt-out

### Does the "User Engagement Data" contain (potentially proprietary) code snippets?

**Comment by davecheney** [comment, marked as answer]
> Does the "User Engagement Data" contain (potentially proprietary) code snippets?

No.

**Reply by Cattes** [reply]
Thank you @davecheney 

Do you also know the answer to

> Does the data collection opt-out also opt-out of "User Engagement Data"?

**Reply by davecheney** [reply]
No.

**Reply by Cattes** [reply]
alright. Thank you for your help.

Maybe someone else could answer the question, if the data collection opt-out is also an opt-out of collecting "User Engagement Data"

**Reply by davecheney** [reply]
When you opt out copilot will not collect your prompts (the source code we feed to the ai model) or its responses.  We do keep “operational” data, which is the record of the request, how long it took, stuff like that which triggers pagers for the on call folks.

**Reply by Cattes** [reply]
Thank you for the explanation @davecheney 

This answers all my questions :+1:
